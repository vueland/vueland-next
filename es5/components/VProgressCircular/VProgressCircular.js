"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.VProgressCircular = void 0;

require("../../../src/components/VProgressCircular/VProgressCircular.scss");

var _vue = require("vue");

var _useColors2 = require("../../effects/use-colors");

var _helpers = require("../../helpers");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var VProgressCircular = (0, _vue.defineComponent)({
  name: 'v-progress-circular',
  props: _objectSpread({
    indeterminate: Boolean,
    size: {
      type: [Number, String],
      "default": 32
    },
    width: {
      type: [Number, String],
      "default": 4
    },
    value: {
      type: [Number, String],
      "default": 0
    },
    rotate: {
      type: [Number, String],
      "default": 0
    }
  }, (0, _useColors2.colorProps)()),
  setup: function setup(props, _ref) {
    var slots = _ref.slots;
    var radius = 20;

    var _useColors = (0, _useColors2.useColors)(),
        setTextColor = _useColors.setTextColor;

    var normalizedValue = (0, _vue.computed)(function () {
      if (props.value < 0) return 0;
      if (props.value > 100) return 100;
      return parseFloat(props.value);
    });
    var classes = (0, _vue.computed)(function () {
      return {
        'v-progress-circular': true,
        'v-progress-circular--indeterminate': props.indeterminate
      };
    });
    var circumference = (0, _vue.computed)(function () {
      return 2 * Math.PI * radius;
    });
    var strokeDashArray = (0, _vue.computed)(function () {
      return Math.round(circumference.value * 1000) / 1000;
    });
    var strokeDashOffset = (0, _vue.computed)(function () {
      return (100 - normalizedValue.value) / 100 * circumference.value + 'px';
    });
    var viewBoxSize = (0, _vue.computed)(function () {
      return radius / (1 - Number(props.width) / +props.size);
    });
    var strokeWidth = (0, _vue.computed)(function () {
      return Number(props.width) / +props.size * viewBoxSize.value * 2;
    });
    var styles = (0, _vue.computed)(function () {
      return {
        width: (0, _helpers.convertToUnit)(props.size),
        height: (0, _helpers.convertToUnit)(props.size)
      };
    });
    var svgStyle = (0, _vue.computed)(function () {
      return {
        transform: "rotate(".concat(+props.rotate, "deg)")
      };
    });

    function genCircle(name, offset) {
      return (0, _vue.h)('circle', {
        "class": "v-progress-circular__".concat(name),
        fill: 'transparent',
        cx: 2 * viewBoxSize.value,
        cy: 2 * viewBoxSize.value,
        r: radius,
        'stroke-width': strokeWidth.value,
        'stroke-dasharray': strokeDashArray.value,
        'stroke-dashoffset': offset
      });
    }

    function genSvg() {
      var children = [props.indeterminate || genCircle('underlay', 0), genCircle('overlay', strokeDashOffset.value)];
      var propsData = {
        xmlns: 'http://www.w3.org/2000/svg',
        viewBox: "".concat(viewBoxSize.value, " ").concat(viewBoxSize.value, " ").concat(2 * viewBoxSize.value, " ").concat(2 * viewBoxSize.value),
        style: svgStyle.value
      };
      return (0, _vue.h)('svg', propsData, children);
    }

    function genInfo() {
      var propsData = {
        "class": 'v-progress-circular__info'
      };
      return (0, _vue.h)('div', propsData, slots["default"] && slots["default"]());
    }

    return function () {
      var propsData = setTextColor(props.color, {
        "class": classes.value,
        style: styles.value
      });
      return (0, _vue.h)('div', propsData, [genSvg(), genInfo()]);
    };
  }
});
exports.VProgressCircular = VProgressCircular;
//# sourceMappingURL=VProgressCircular.js.map