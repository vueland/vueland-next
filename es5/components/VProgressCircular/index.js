"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _VProgressCircular = require("./VProgressCircular");

Object.keys(_VProgressCircular).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (key in exports && exports[key] === _VProgressCircular[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _VProgressCircular[key];
    }
  });
});
//# sourceMappingURL=index.js.map