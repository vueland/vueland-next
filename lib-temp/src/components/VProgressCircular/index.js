export * from './VProgressCircular';
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi8uLi8uLi9zcmMvY29tcG9uZW50cy9WUHJvZ3Jlc3NDaXJjdWxhci9pbmRleC50cyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQSxjQUFjLHFCQUFxQixDQUFBIiwic291cmNlc0NvbnRlbnQiOlsiZXhwb3J0ICogZnJvbSAnLi9WUHJvZ3Jlc3NDaXJjdWxhcidcbiJdfQ==