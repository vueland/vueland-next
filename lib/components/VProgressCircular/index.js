export * from './VProgressCircular';
//# sourceMappingURL=index.js.map