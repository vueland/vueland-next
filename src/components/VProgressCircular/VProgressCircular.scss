@import "../../styles/scss/modules/global";
@import "variables";

.v-progress-circular {
  position: relative;
  display: inline-flex;
  vertical-align: middle;
  justify-content: center;
  align-items: center;

  & > svg {
    width: 100%;
    height: 100%;
    margin: auto;
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    z-index: 0;
  }

  &--indeterminate {
    > svg {
      animation: $progress-circular-rotate-animation;
      transition: $progress-circular-indeterminate-transition;
      transform-origin: center center;
    }

    .v-progress-circular__overlay {
      animation: $progress-circular-dash-animation;
      stroke-linecap: round;
      stroke-dasharray: 80, 200;
      stroke-dashoffset: 0;
    }
  }

  &__overlay {
    stroke: currentColor;
    transition: $progress-circular-overlay-transition
  }

  &__underlay {
    stroke: map-get($grey, 'lighten-3');
  }

  &__info {
    align-items: center;
    display: flex;
    justify-content: center;
  }
}

@keyframes progress-circular-dash {
  0% {
    stroke-dasharray: 1, 200;
    stroke-dashoffset: 0;
  }
  50% {
    stroke-dasharray: 100, 200;
    stroke-dashoffset: -15px;
  }
  100% {
    stroke-dasharray: 100, 200;
    stroke-dashoffset: -125px;
  }
}

@keyframes progress-circular-rotate {
  100% {
    transform: rotate(360deg);
  }
}

