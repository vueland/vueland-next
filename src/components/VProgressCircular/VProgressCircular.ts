import { h, computed, defineComponent } from 'vue'

// Effects
import { colorProps, useColors } from '../../composable/use-colors'

// Types
import { VNode } from 'vue'
import { convertToUnit } from '../../helpers'

export const VProgressCircular = defineComponent({
  name: 'v-progress-circular',
  props: {
    indeterminate: Boolean,
    size: {
      type: [Number, String],
      default: 32,
    },
    width: {
      type: [Number, String],
      default: 4,
    },
    value: {
      type: [Number, String],
      default: 0,
    },
    rotate: {
      type: [Number, String],
      default: 0,
    },
    ...colorProps(),
  },
  setup(props, { slots }) {
    const radius = 20

    const { setTextClassNameColor, setTextCssColor } = useColors()

    const normalizedValue = computed<number>(() => {
      if (props.value < 0) return 0

      if (props.value > 100) return 100

      return parseFloat(props.value as string)
    })

    const classes = computed<Record<string, boolean>>(() => ({
      'v-progress-circular': true,
      'v-progress-circular--indeterminate': props.indeterminate,
      ...(props.color ? setTextClassNameColor(props.color) : {}),
    }))

    const circumference = computed<number>(() => {
      return 2 * Math.PI * radius
    })

    const strokeDashArray = computed<number>(() => {
      return Math.round(circumference.value * 1000) / 1000
    })

    const strokeDashOffset = computed<string>(() => {
      return ((100 - normalizedValue.value) / 100) * circumference.value + 'px'
    })

    const viewBoxSize = computed<number>(() => {
      return radius / (1 - Number(props.width) / +props.size)
    })

    const strokeWidth = computed<number>(() => {
      return (Number(props.width) / +props.size) * viewBoxSize.value * 2
    })

    const styles = computed<object>(() => ({
      width: convertToUnit(props.size),
      height: convertToUnit(props.size),
      ...(props.color ? setTextCssColor(props.color) : {}),
    }))

    const svgStyle = computed(() => {
      return {
        transform: `rotate(${+props.rotate}deg)`,
      }
    })

    function genCircle(name: string, offset: string | number): VNode {
      return h('circle', {
        class: `v-progress-circular__${name}`,
        fill: 'transparent',
        cx: 2 * viewBoxSize.value,
        cy: 2 * viewBoxSize.value,
        r: radius,
        'stroke-width': strokeWidth.value,
        'stroke-dasharray': strokeDashArray.value,
        'stroke-dashoffset': offset,
      })
    }

    const genSvg = () => {
      const children = [
        props.indeterminate || genCircle('underlay', 0),
        genCircle('overlay', strokeDashOffset.value),
      ]

      const propsData = {
        xmlns: 'http://www.w3.org/2000/svg',
        viewBox: `${viewBoxSize.value} ${viewBoxSize.value} ${
          2 * viewBoxSize.value
        } ${2 * viewBoxSize.value}`,

        style: svgStyle.value,
      }

      return h('svg', propsData, children)
    }

    const genInfo = () => {
      const propsData = {
        class: 'v-progress-circular__info',
      }
      return h('div', propsData, slots.default && slots.default())
    }

    return () => {
      const propsData = {
        class: classes.value,
        style: styles.value,
      }
      return h('div', propsData, [genSvg(), genInfo()])
    }
  },
})
