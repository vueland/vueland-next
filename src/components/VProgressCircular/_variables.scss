$progress-circular-dash-animation: progress-circular-dash 1.4s ease-in-out infinite;
$progress-circular-rotate-animation: progress-circular-rotate 1.4s linear infinite;
$progress-circular-overlay-transition: stroke-dashoffset .3s ease-in-out;
$progress-circular-indeterminate-transition: all .6s ease-in-out
