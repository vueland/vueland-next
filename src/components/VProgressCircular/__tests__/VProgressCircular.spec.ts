import { mount } from '@vue/test-utils'
import { VProgressCircular } from '../VProgressCircular'

describe('VProgressCircular', () => {
  let mountFunction

  mountFunction = (options = {}) => mount(VProgressCircular, { ...options })

  it('should mount component and match snapshot', () => {
    const cmp = mountFunction()

    expect(cmp.html()).toMatchSnapshot()
  })
})
