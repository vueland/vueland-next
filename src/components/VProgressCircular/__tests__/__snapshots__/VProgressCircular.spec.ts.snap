// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VProgressCircular should mount component and match snapshot 1`] = `
<div class="v-progress-circular"
     style="width: 32px; height: 32px;"
>
  <svg xmlns="http://www.w3.org/2000/svg"
       viewbox="22.857142857142858 22.857142857142858 45.714285714285715 45.714285714285715"
       style="transform: rotate(0deg);"
  >
    <circle class="v-progress-circular__underlay"
            fill="transparent"
            cx="45.714285714285715"
            cy="45.714285714285715"
            r="20"
            stroke-width="5.714285714285714"
            stroke-dasharray="125.664"
            stroke-dashoffset="0"
    >
    </circle>
    <circle class="v-progress-circular__overlay"
            fill="transparent"
            cx="45.714285714285715"
            cy="45.714285714285715"
            r="20"
            stroke-width="5.714285714285714"
            stroke-dasharray="125.664"
            stroke-dashoffset="125.66370614359172px"
    >
    </circle>
  </svg>
  <div class="v-progress-circular__info">
  </div>
</div>
`;
